version: 1
style: integration
description: must properly handle auth error on check out
runOn:
  -
    # required for appName in fail point
    minServerVersion: "4.9.0"
    auth: true
failPoint:
  configureFailPoint: failCommand
  mode: { times: 1 }
  data:
    failCommands: ["saslContinue"]
    appName: &appName "poolCheckOutAuthErrorTest"
    errorCode: 18
poolOptions:
  appName: *appName
operations:
  - name: ready
  - name: checkOut
error:
  type: AuthenticationError
events:
  - type: ConnectionCheckOutStarted
    address: 42
  - type: ConnectionCreated
    address: 42
  - type: ConnectionPoolCleared
    address: 42
  - type: ConnectionClosed
    address: 42
    reason: error
  - type: ConnectionCheckOutFailed
    address: 42
    connectionId: 42
    reason: connectionError
ignore:
  - ConnectionPoolCreated
  - ConnectionPoolReady
